packages/core/src/browser/common-frontend-contribution.ts

```diff
@@ -2238,4 +2245,164 @@ export class CommonFrontendContribution implements FrontendApplicationContributi
             }
         );
     }
+
+    registerThemeStyle(theme: ColorTheme, collector: CssStyleCollector): void {
```

Comment:
This aggregates a lot of styling that it might be nice to do more locally, both for legibility and overridability (since overriding this method would likely mean copying it, though I guess you could also just call `super` and then add the rules you wanted later in the same sheet). One easy to implement but harder to override idea might be something like a static `registerThemeStyle` method on, say, `TreeWidget` and then a registration `bind(StylingParticipant).toConstantValue(TreeWidget);`; a heavier but easier to override alternative would be adding a little `TreeWidgetStylingParticipant` service (and `StatusBarStylingParticipant` etc. etc.). What do you think about moving some of the styling closer to the affected elements, and what the best way to do that would be?